client/state/viewers/reducer.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { keyBy, omit, union } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	REMOVE_VIEWER_SUCCESS,
+	VIEWERS_REQUEST,
+	VIEWERS_REQUEST_FAILURE,
+	VIEWERS_REQUEST_SUCCESS,
+} from 'calypso/state/action-types';
+import { combineReducers } from 'calypso/state/utils';
+
+export const items = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case VIEWERS_REQUEST_SUCCESS: {
+			return Object.assign(
+				{},
+				state,
+				keyBy( action.data?.viewers, ( viewer ) => viewer.ID )
+			);
+		}
+		case REMOVE_VIEWER_SUCCESS: {
+			return Object.assign( {}, omit( state, action.viewerId ) );
```

Comment:
Hm, we're dealing with an object in this case so `Array.filter` wouldn't work.

Reply:
Then maybe `Object.fromEntries( Object.entries().filter( ... ) )`?

Reply:
You could do something like:

```js
return Object.fromEntries( Object.entries( state ).filter( ( [ key, value ] ) => key !== action.viewerId ) );
```

but in my opinion that might be best left for a future Lodash removal PR. It would probably be best wrapped in a utility method, too.

Reply:
Yep, as @sgomes said, this isn't a blocker, just a personal nitpick.

Reply:
> It would probably be best wrapped in a utility method, too.

:+1: `omit` is one of these functions that don't have a 1:1 ESnext equivalent. Conceptually, it belongs to a library of "immutability helpers", as it removes a property from an object immutably. Similar to the mutating `delete state[ action.viewerId ]`.